### PR TITLE
Move from app.php to Application.php

### PR DESCRIPTION
**Pull request by Altahrim**
Use event `OCP\User\Events\BeforeUserLoggedInEvent`

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "nextcloud/limit_login_to_ip",
 	"description": "Lint config for nextcloud/limit_login_to_ip",
-	"license": "MIT",
```
Comment (miaulalala) [inline composer.json:4]: Not sure about simply changing the license?

Comment (Altahrim) [reply composer.json:4]: You are right.
I believe I saw AGPL v3 somewhere so I wanted to make it consistent, but I don't find it again.
In any case, it should not be in this pull request :+1:

File: lib/AppInfo/Application.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2017 Lukas Reschke <lukas@statuscode.ch>
```
Comment (miaulalala) [inline lib/AppInfo/Application.php:6]: wasn't Lukas ;)

File: lib/IsRequestAllowed.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2017 Lukas Reschke <lukas@statuscode.ch>
```
Comment (miaulalala) [inline lib/IsRequestAllowed.php:5]: same

**Review by miaulalala — APPROVED** [review]
LGTM, didn't test

File: lib/AppInfo/Application.php (hunk 2 above)
Comment (Altahrim) [reply lib/AppInfo/Application.php:6]: Haha, yes ^^

File: lib/Controller/LoginDeniedController.php [hunk 4]
```diff
@@ -27,13 +27,15 @@
 use OCP\AppFramework\Http;
 use OCP\AppFramework\Http\TemplateResponse;
 
+/**
+ * @api
```
Comment (nickvergessen) [inline lib/Controller/LoginDeniedController.php:31]: What is this about?

Comment (Altahrim) [reply lib/Controller/LoginDeniedController.php:31]: `psalm` complaining the class isn't used. 
There is also `@psalm-api` which is maybe more explicit :)

**Pull request merged**